blog/_posts/2022-05-03-vscode-extension.md

```diff
@@ -0,0 +1,107 @@
+---
+slug: "vscode-extension"
+date: 2022-05-03
+author: siegs
+layout: blog
+title: A VS Code Extension for Tiltfiles
+image: /assets/images/vscode-extension/gears.jpg
+image_caption: 'Plastic Gear Lot from <a href="https://www.pexels.com/photo/assorted-color-plastic-gear-lot-1476320/">Dan Cristian Pădureț</a>'
+subtitle: Improve Tiltfile authoring when working in VS Code
+description: Stop switching between code and documentation and use our VS Code Extension to write your Tiltfiles!
+tags:
+- tiltfile
+- devtools
+- vscode
+---
+
+If you use [Visual Studio Code][code], are new to Tilt and have been spending a lot of time switching between writing your `Tiltfile` and reading our [`Tiltfile` API Reference][docs], stop! Go [install our new `Tiltfile` extension][install] and let us help you write `Tiltfile`s more easily! And if you're not a fan of VS Code, stay to learn about the components we used to build the extension. (You might be able to re-use what we built into Tilt to drive a `Tiltfile` extension in your own editor!)
+
+Our `Tiltfile` extension is [available in the Visual Studio marketplace][install], and can also be found in the Extensions sidebar inside VS Code. The extension requires you to have Tilt installed already; make sure you're using [the latest release][upgrade] to ensure you have all the features available.
+
+## Syntax Highlighting
+
+`Tiltfile`s are based on [Starlark][starlark], a dialect of Python. The extension automatically highlights Starlark thanks to a TextMate bundle derived from [Bazel's VS Code Extension][Bazel] and [Magic Python][].
+
+You can also use [our TextMate bundle for syntax highlighting with other IDEs that support it][tmbundle].
+
+## Completion and Signature Help
+
+A major reason for using an IDE like VS Code is to help you write code faster and more correctly with context-aware hints. The `Tiltfile` extension adds symbol completion for [Starlark builtin functions and methods][starlark], [Tilt builtins][docs], and any variables or functions you define in your `Tiltfile`s. The documentation shown in pop-ups is sourced from the same code that generates [our API docs][docs].
```

Comment:
I use the word "symbol" here and once or twice below; should I use "name" or "variable name" instead? "Symbol" feels more like unnecessary compiler-speak.

Reply:
probably not worth worrying about - I think "completion" and the fact that the sentence then lists all the things completed do enough work.

FWIW, I checked a few language extensions in the marketplace and they variously say "symbols", "Intellisense", and "methods and variables".